tests/lib/helpers.js

```diff
@@ -2,20 +2,72 @@
  * http://creativecommons.org/publicdomain/zero/1.0/ */
 
 const webpack = require("webpack");
-// import webpack from "webpack";
+const CopyWebpackPlugin = require("copy-webpack-plugin");
+const version = require("../../package.json").version;
 
 /*
   Returns a promise that resolves once webpack has compiled all the addon
   configs have been built.
 */
-// export default function
 
 function compileWebpack() {
   return new Promise((resolve, reject) => {
-    const configs = ["chrome", "firefox", "firefox-mobile", "opera"].map(item =>
-      require(`../../${item}/webpack.config.js`)
+    const config = ["chrome", "firefox", "firefox-mobile", "opera"].map(
+      platform => {
+        var template = {
+          entry: `./${platform}/addon.js`,
+          output: {
+            filename: `../dist/${platform}/background.js`
+          },
+          plugins: [
+            new CopyWebpackPlugin([
+              {
+                from: "shared/content.js",
+                to: `../dist/${platform}/content.js`
+              },
+              {
+                from: "shared/*.png",
+                to: `../dist/${platform}/[name].[ext]`
+              },
+              {
+                from: "shared/manifest.json",
+                to: `../dist/${platform}/[name].json`,
+                transform: function(content, path) {
+                  let manifest = JSON.parse(content);
+                  // Derive addon versioning from package.json
+                  manifest["version"] = version;
+                  switch (platform) {
+                    case "firefox":
+                      // Add Firefox-specific bits to the manifest.json
+                      manifest["applications"] = {
+                        gecko: {
+                          id: "jid1-mjpB54bRzP9Zxw@jetpack",
```

Comment:
unrelated: i really hate this legacy jetpack id. but if we change it, it means no more automatic upgrades. 😢